hazelcast/src/main/java/com/hazelcast/config/MapConfig.java

```diff
@@ -674,15 +679,68 @@ public boolean isNearCacheEnabled() {
         return nearCacheConfig != null;
     }
 
+    /**
+     *
+     * @return
+     * @deprecated user {@link #getQueryCacheConfigs()} instead.
+     */
     public boolean isOptimizeQueries() {
-        return optimizeQueries;
+        return cacheDeserializedValues == CacheDeserializedValues.ALWAYS;
     }
 
+    /**
+     *
+     * Enable de-serialized value caching when evaluating predicates. It has no effect when {@link InMemoryFormat}
+     * is {@link InMemoryFormat#OBJECT} or when {@link com.hazelcast.nio.serialization.Portable} serialization is used.
+     *
+     * @param optimizeQueries
+     * @return this {@code MapConfig} instance.
+     * @deprecated use {@link #setCacheDeserializedValues(CacheDeserializedValues)} instead
+     * @see {@link CacheDeserializedValues}
+     */
     public MapConfig setOptimizeQueries(boolean optimizeQueries) {
-        this.optimizeQueries = optimizeQueries;
+        if (optimizeQueries) {
+            if (cacheDeserializedValues == DEFAULT_CACHED_DESERIALIZED_VALUES) {
+                cacheDeserializedValues = CacheDeserializedValues.ALWAYS;
+            } else if (cacheDeserializedValues == CacheDeserializedValues.NEVER) {
+                throw new ConfigurationException("Deprecated option 'optimize-queries' is set to true, "
+                        + "but 'cacheDeserializedValues' is set to NEVER. These are conflicting options. Please "
+                        + "remove the `optimize-queries'");
+            }
+        } else {
+            if (cacheDeserializedValues == CacheDeserializedValues.ALWAYS) {
+                throw new ConfigurationException("Deprecated option 'optimize-queries' is set to false, "
+                        + "but 'cacheDeserializedValues' is set to ALWAYS. These are conflicting options. Please "
+                        + "remove the `optimize-queries'");
+            }
+        }
         return this;
```

Comment:
this is a bit tricky since I have removed the `optimizeQuery` flag. The `setOptimizeQueries()` actually set the `cacheDeserializedValues` under the hood. The beauty of encapsulation.


Any I might re-introduce the `optimizeQuery` flag just to help with conflict detection. WDYT?